File: GILua/src/main/java/org/anime_game_servers/gi_lua/models/scene/group/SceneNPC.java
package org.anime_game_servers.gi_lua.models.scene.group;

import lombok.Getter;
import lombok.ToString;
import org.anime_game_servers.core.base.annotations.lua.LuaNames;
import org.anime_game_servers.gi_lua.models.constants.EntityType;

@ToString
@Getter
public class SceneNPC extends SceneObject{
    @LuaNames("npc_id")
	private int npc_id;

    @Override
    public EntityType getType() {
        return EntityType.NPC;
    }
}

File: GILua/src/main/java/org/anime_game_servers/gi_lua/models/scene/group/SceneNPC.kt
package org.anime_game_servers.gi_lua.models.scene.group

import org.anime_game_servers.core.base.annotations.lua.LuaNames
import org.anime_game_servers.gi_lua.models.constants.EntityType

data class SceneNPC(
    @field:LuaNames("npc_id")
    val npcId: Int = 0,
    val room: Int = 0,
    val pointID: Int = 0,
    ) : SceneObject() {
    override val type: EntityType = NPC
}
